components/post_view/combined_system_message/combined_system_message.jsx

```diff
@@ -294,35 +294,50 @@ export default class CombinedSystemMessage extends React.PureComponent {
         );
     }
 
+    renderMessage(postType, userIds, actorId) {
+        return (
+            <React.Fragment key={postType + actorId}>
+                {this.renderFormattedMessage(postType, userIds, actorId)}
+                <br/>
+            </React.Fragment>
+        );
+    }
+
     render() {
-        const {messageData} = this.props;
+        const {
+            currentUserId,
+            messageData,
+        } = this.props;
 
         const content = [];
+        const removedUserIds = [];
         for (const message of messageData) {
             const {
                 postType,
                 actorId,
             } = message;
             let userIds = message.userIds;
 
-            if (!this.props.showJoinLeave && actorId !== this.props.currentUserId) {
-                const affectsCurrentUser = userIds.indexOf(this.props.currentUserId) !== -1;
+            if (!this.props.showJoinLeave && actorId !== currentUserId) {
+                const affectsCurrentUser = userIds.indexOf(currentUserId) !== -1;
 
                 if (affectsCurrentUser) {
                     // Only show the message that the current user was added, etc
-                    userIds = [this.props.currentUserId];
+                    userIds = [currentUserId];
                 } else {
                     // Not something the current user did or was affected by
                     continue;
                 }
+            } else if (postType === REMOVE_FROM_CHANNEL) {
```

Comment:
Same thought as [here](https://github.com/mattermost/mattermost-mobile/pull/1955#discussion_r207383224)

Reply:
Oh, you're right (although still need the else-if).  Not really need but I update the commit of mattermost-redux.
Done update.  Hopefully, all is good and covered.  Thanks!

Reply:
Those changes do fix the problem, but removing the else-if would make it much simpler since we wouldn't be using `continue;` to break the control flow as often.
```javascript
if (!this.props.showJoinLeave && actorId !== currentUserId) {
    const affectsCurrentUser = userIds.indexOf(currentUserId) !== -1;

    if (affectsCurrentUser) {
        // For a REMOVE_FROM_CHANNEL message, this will still get hit, setting the userIds to only be the current user
        userIds = [currentUserId];
    } else {
        continue;
    }
}

if (postType === REMOVE_FROM_CHANNEL) {
    // And then it'll reach here, update removedUserIds with the current user, and then continue the loop
    removedUserIds.push(...userIds);
    continue;
}
```

Reply:
Thanks, that's neat!
Did the quick test and it works great, also the previous test in placed didn't break :smile: